Where_Is_My_Wife/Assets/_Main/Assets/Saved_system/SaveData.cs

```diff
@@ -0,0 +1,114 @@
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace WhereIsMyWife.SavedData
+{
+    public class SavedData : MonoBehaviour
+    {
+        [System.Serializable]
+        public class SavedDataInfo
+        {
+            public int LastLevelFinished { get; private set; }
+            public Dictionary<string, object> Settings { get; private set; }
+            public List<int> CompletedLevels { get; private set; }
+
+            [JsonConstructor]
+            public SavedDataInfo(int lastLevelFinished, Dictionary<string, object> settings, List<int> completedLevels)
+            {
+                LastLevelFinished = lastLevelFinished;
+                Settings = settings ?? new Dictionary<string, object>();
+                CompletedLevels = completedLevels ?? new List<int>();
+            }
+
+            public void SetLastLevel(int level) => LastLevelFinished = level;
+
+            public void AddCompletedLevel(int level)
+            {
+                if (!CompletedLevels.Contains(level))
+                {
+                    CompletedLevels.Add(level);
+                    Debug.Log($"Level {level} marked as completed.");
+                }
+            }
+        }
+
+        private const string SaveKey = "saved-data";
+        private SavedDataInfo savedData;
+
+        [Header("Save System Buttons")] [SerializeField]
+        private Button saveButton;
+
+        [SerializeField] private Button loadButton;
+        [SerializeField] private Button deleteButton;
+
+        private void Start()
+        {
+            Load();
+            AssignButtonListeners();
+        }
+
+        private void AssignButtonListeners()
+        {
+            saveButton?.onClick.AddListener(Save);
+            loadButton?.onClick.AddListener(Load);
+            deleteButton?.onClick.AddListener(DeleteSave);
+        }
+
+        public void CompleteLevel(int level)
+        {
+            if (savedData == null)
+            {
+                savedData = new SavedDataInfo(level, new Dictionary<string, object>(), new List<int>());
+            }
+            else
+            {
+                savedData.SetLastLevel(level);
+                savedData.AddCompletedLevel(level);
```

Comment:
This should only set the last level completed, an int, the completed level is not needed